opendevin/runtime/utils/runtime_build.py

```diff
@@ -105,30 +109,30 @@ def _generate_dockerfile(
     dockerfile_content += (
         'RUN if [ -d /opendevin/code ]; then rm -rf /opendevin/code; fi\n'
     )
-    # Unzip the tarball to /opendevin/code
+    # Unzip the tarball to /opendevin/code and install dependencies via poetry
     dockerfile_content += (
         'RUN cd /opendevin && tar -xzvf project.tar.gz && rm project.tar.gz\n'
-    )
-    dockerfile_content += f'RUN mv /opendevin/{source_code_dirname} /opendevin/code\n'
-
-    # ALTERNATIVE, but maybe not complete? (toml error!)
-    dockerfile_content += (
-        'RUN cd /opendevin/code && '
-        '/opendevin/miniforge3/bin/mamba run -n base poetry env use python3.11 && '
-        '/opendevin/miniforge3/bin/mamba run -n base poetry install --no-interaction --no-root\n'
-        'RUN /opendevin/miniforge3/bin/mamba run -n base poetry cache clear --all . && \\\n'
-        'apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* &&\\\n'
-        '/opendevin/miniforge3/bin/mamba clean --all\n'
+        f'RUN mv /opendevin/{source_code_dirname} /opendevin/code\n'
+        'WORKDIR /opendevin/code\n'
+        'RUN /opendevin/miniforge3/bin/mamba run -n base poetry env use python3.11\n'
+        # Activate the opendevin virtual environment
+        'RUN source /opendevin/code/.venv/bin/activate\n'
+        'RUN /opendevin/miniforge3/bin/mamba run poetry install --only main --no-interaction --no-root\n'  # -n base
     )
 
     # For browser (update if needed)
     dockerfile_content += (
         'RUN apt-get update && \\\n'
-        '    cd /opendevin/code && \\\n'
-        '    /opendevin/miniforge3/bin/mamba run -n base poetry run pip install playwright && \\\n'
-        '    /opendevin/miniforge3/bin/mamba run -n base poetry run playwright install --with-deps chromium && \\\n'
-        '    apt-get clean && \\\n'
-        '    rm -rf /var/lib/apt/lists/*\n'
+        '  /opendevin/miniforge3/bin/mamba run -n base poetry run pip install --upgrade pip &&\\\n'
+        '  /opendevin/miniforge3/bin/mamba run -n base poetry run pip install playwright'
+        ' jupyterlab notebook jupyter_kernel_gateway flake8 '
+        ' python-docx PyPDF2 python-pptx pylatexenc openai opencv_python'
+        ' diskcache grep-ast tree-sitter pathspec'
+        ' python-dotenv toml pydantic python-docx pyyaml '
+        ' docker pexpect tenacity e2b minio &&\\\n'
```

Comment:
plus, i don't think we need (and should) keep all these packages explicitly here, since they are all already in the `pyproject.toml`.

Reply:
In my tests the plugins failed in the container as something was missing. 🤔